metabase/metabase
# fix(sdk): Standardize SDK components errors with different ID formats

Closes https://github.com/metabase/metabase/issues/49581

[Error component design (Figma)](https://www.figma.com/design/W1TfzdrR8ldzp4vdjP1z36/SDK-Embedding%3A-admin-settings-%2B-embed-flow?node-id=11613-23031&t=z4vjXKUradcAyyAp-0)

```[tasklist]
### Todo
- [x] Extract color in the `Code` component that is for displaying errored ID to `colors.module.css`
- [x] Clean up storybook (if we want to keep the ID arguments) or just revert them.
```


### Description

Make all question/dashboard public components loading/error state consistent.

The design is subject to change.

I haven't thought much how to deal with stories, I think each story's control has value. Please let me know if you have any idea how to keep those stories, or maybe we don't need them after the development since we already have Cypress tests for them now. Thought?

### How to verify

Go to each story below and follow the same instructions.
- InteractiveQuestion.stories.tsx
- StaticQuestion.stories.tsx
- InteractiveDashboard.stories.tsx
- StaticDashboard.stories.tsx
- EditableDashboard.stories.tsx

To test each story:
1. Open the storybook's Default story.
2. Change each question/dashboard ID to each option in the addon section
    ![image](https://github.com/user-attachments/assets/5e50153b-99b9-4bba-b5ad-15dbeb2b318b)
3. The result should match the expected behavior table in #49581

### Demo
![image](https://github.com/user-attachments/assets/c1852095-38ff-491c-b816-c416d74e76f9)


### Checklist
